pilopt/src/lib.rs

```diff
@@ -565,6 +565,8 @@ fn remove_trivial_identities<T: FieldElement>(pil_file: &mut Analyzed<T>) {
                 assert_eq!(left.expressions.len(), right.expressions.len());
                 left.expressions.is_empty().then_some(index)
             }
+            Identity::PhantomLookup(..) => None,
+            Identity::PhantomPermutation(..) => None,
             Identity::Permutation(..) => None,
```

Comment:
The above rule would apply to these 3 as well, no?

Reply:
Oh yes!

Reply:
done